lib/matplotlib/collections.py

```diff
@@ -258,13 +258,7 @@ def get_datalim(self, transData):
 
         transform = self.get_transform()
         offset_trf = self.get_offset_transform()
-        has_offsets = np.any(self._offsets)  # True if any non-zero offsets
-        if has_offsets and not offset_trf.contains_branch(transData):
-            # if there are offsets but in some coords other than data,
-            # then don't use them for autoscaling.
-            return transforms.Bbox.null()
```

Comment:
I am very wary of just ripping out a block of code like this without justification. The original issue was that the autoscaling logic was faulty when the offsets was basically null. But this block is for a special case where the offset transform isn't the usual transform and that we can't usually calculate the autoscale correctly in that case. Just ripping this out doesn't address this special case.

Reply:
I am also highly suspicious that ripping this block out did not trigger a test failure

Reply:
Thanks for the push to investigate this, Ben. You were right to be wary of this passing without any tests.

I added a new test that adds a scatter in transAxes coordinates (fixing the point in Axes space). Which I think we agree should not contribute to autoscaling. When I ripped this chunk of code out, it _would_ have contributed.

I did however change the conditional to be: "if offsets is an updated transform (not Identity), and does not contain the transData path". Rather than checking for that possibly Falsy "has_offsets".